### ISSUE TITLE
Toonz Raster Level Merging Palette/Style handling (still) buggy, code hint included

### ISSUE THREAD

**Issue by Bracket-H** [opentoonz/opentoonz#4133]
This issue, seems to be still an issue, at least for me.
https://github.com/opentoonz/opentoonz/issues/1670

Where merged raster TLVs only merge and adjust styles correctly for the first cell of the level.

I'm not versed in the codebase, but I did do a preliminary hack which resolves this issue, but only in a specific, roundabout way that might not be applicable to regular use since it relies on a studio palette.

Code hint:
```diff
diff --git a/toonz/sources/toonz/mergecmapped.cpp b/toonz/sources/toonz/mergecmapped.cpp
index 20378b64..a2fe9c58 100644
--- a/toonz/sources/toonz/mergecmapped.cpp
+++ b/toonz/sources/toonz/mergecmapped.cpp
@@ -58,17 +58,17 @@ public:
 void mergeCmapped(const std::vector<MergeCmappedPair> &matchingLevels) {
   if (matchingLevels.empty()) return;
 
-  TPalette *palette = matchingLevels[0].m_cell->getImage(false)->getPalette();
-  TPalette *matchPalette =
-      matchingLevels[0].m_mcell->getImage(false)->getPalette();
-
-  TPalette::Page *page;
 
   // upInkId -> downInkId
   std::map<int, int> usedColors;
 
   int i = 0;
   for (i = 0; i < (int)matchingLevels.size(); i++) {
+  TPalette *palette = matchingLevels[i].m_cell->getImage(false)->getPalette();
+  TPalette *matchPalette =
+      matchingLevels[i].m_mcell->getImage(false)->getPalette();
+
+  TPalette::Page *page;
     TToonzImageP img = (TToonzImageP)matchingLevels[i].m_cell->getImage(true);
     TToonzImageP match =
         (TToonzImageP)matchingLevels[i].m_mcell->getImage(false);

```
This change alone correctly merges any amount of TLVs, with correct style indices and everything.
However, this change also duplicates styles since it doesn't merge/adjust the inks/fills in the pixel data of the cells themselves, it just uses all the different indices of the original linework, or something like that (lack the skill/knowledge of tracing that right now)

The workaround is to have a studio palette that contains all the colors/styles used in the final merged level.
One can then, through the palette browser, "adjust the levels palette with the studio palette", and choosing a tolerance of '0', will clean up the palette of the merged level, as well as consolidating/merging all the inks and fills, collapsing the indices to the shared '0 tolerance' final value.

In other words, ink 6,12,44,100 all with the same style value (ie, they're all the same color they just have different indices), get  collapsed into a single index.

I hope that makes sense.

I realize that this might: 
not always be wanted, just because a color is shared, doesn't mean the artist intends it as the same index/logical part of the image
might not work without a studio palette or similar to 'kick out' the doubles with.

Bottom line, however is:
The bug is still there. With or without this scenario.

**Comment by Bracket-H** [comment]
Note: This bug occurs even without messing with color index 1, aka the standard solid black style.

**Comment by Bracket-H** [comment]
Somewhat related to this, but also a bit off topic since it's a feature request, really:

I often wish OpenToonz had a way to fix/renumber indices for an active level itself, that adjusts pixel data at the same time.

For example, after doing some work, a level ends up using style index 0,1,2,5,8,9

If, due some operation now, index 5 got deleted, but the lines/filles remain, and the palette ends up looking like 0,1,2,8,9
Then fixing this right now (unless there's a way) is akward.

Either one adds styles, which, through the process of adding, get filled into the 'gaps' of the indices, until one arrives at index 5 (with index 3 and 4 being extraneous/irrelevant at that point).
Or, one uses the painroller thing that dyes existing pixels with a new value.

It would be nicer if one could just go: What before was index 5, now shall be index 3, and all the pixels of 5 get adjusted to 3.

**Comment by RodneyBaker** [comment]
@Bracket-H 
I know this report is a little dated without any feedback.
You've got some interesting thoughts here and I hope they can be discussed in more depth.

Thanks by the way for the code hints. 
Much appreciated!
